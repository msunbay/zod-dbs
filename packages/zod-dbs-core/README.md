# zod-dbs-core

Shared core functionality for [zod-dbs](https://github.com/msunbay/zod-dbs).
