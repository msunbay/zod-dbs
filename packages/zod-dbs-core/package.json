{
  "name": "zod-dbs-core",
  "version": "1.0.13",
  "description": "Shared core functionality for zod-dbs packages",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "type": "module",
  "scripts": {
    "build": "rm -rf ./dist && tsc --project tsconfig.build.json",
    "test": "pnpm run test:types && pnpm run test:unit",
    "test:unit": "vitest --project unit run",
    "test:types": "tsc --noEmit"
  },
  "author": "msunbay",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/msunbay/zod-dbs.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "debug": "^4.4.1"
  }
}
