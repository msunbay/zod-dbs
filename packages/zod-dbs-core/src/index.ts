export { getZodType } from './getZodType.js';
export type * from './types.js';
export * from './utils/index.js';
export { ZodDbsBaseProvider } from './ZodDbsBaseProvider.js';
