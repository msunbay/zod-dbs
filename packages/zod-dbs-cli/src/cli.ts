import { generateZodSchemas } from 'zod-dbs';
import { logDebug, toError } from 'zod-dbs-core';

import type { ZodDbsCliConfig, ZodDbsCliOptions } from './types.js';

import { detectDebugMode, getConfiguration } from './config.js';
import { runCommander } from './program.js';
import { loadProvider } from './provider.js';
import { isSilentMode } from './utils/args.js';
import { logAppName, logEmpty, logError, logSetting } from './utils/logger.js';
import { createProgressHandler } from './utils/progress.js';
import { getAppVersion } from './utils/version.js';

export const runCli = async (cliOptions: ZodDbsCliOptions = {}) => {
  detectDebugMode();
  logDebug('Starting zod-dbs CLI with options:', cliOptions);

  const appVersion = cliOptions.appVersion ?? (await getAppVersion());
  const appName = cliOptions.appName || 'zod-dbs';
  const silent = isSilentMode();

  if (!silent) {
    logAppName(`${appName} CLI`);
  }

  const config = await getConfiguration(cliOptions);
  const provider = await loadProvider(cliOptions.overrides?.provider, config);

  const options = runCommander({
    provider,
    appName,
    appVersion,
    includeProviderOption: !cliOptions.overrides?.provider,
  });

  const spinner = createProgressHandler(silent);

  try {
    const cliConfig: ZodDbsCliConfig = {
      ...config,
      ...options,
    };

    logDebug('Final configuration:', cliConfig);

    if (!cliConfig.silent) {
      logSettings(cliConfig);
      logEmpty();
    }

    await generateZodSchemas({
      provider,
      renderer: cliConfig.renderer,
      config: {
        ...cliConfig,
        onProgress: spinner.onProgress,
      },
    });

    spinner.done();
  } catch (error) {
    spinner.fail();

    logError(toError(error).message);
    logDebug(error);

    process.exit(1);
  }
};

const logSettings = (config: ZodDbsCliConfig) => {
  for (const [key, value] of Object.entries(config)) {
    if (value !== undefined && value !== null) {
      logSetting(key, value);
    }
  }
};
