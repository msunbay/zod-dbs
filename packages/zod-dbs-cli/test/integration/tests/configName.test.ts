import { toError } from 'zod-dbs-core';

import type { CliExecutionError } from '../../../../../test/integration/utils/cli.js';

import { executeCli } from '../../../../../test/integration/utils/cli.js';

it('CLI accepts custom config file name', async () => {
  const output = await executeCli(`--config-name test`);

  expect(output).toMatchSnapshot();
});

it('CLI fails if custom config file name does not exist', async () => {
  let error: CliExecutionError | undefined;

  try {
    await executeCli(`--config-name nonexisting`, { logErrors: false });
  } catch (err) {
    error = toError(err);
  }

  expect(error).toBeDefined();
  expect(error!.stderr).toContain(
    'Failed to load configuration from file "zod-dbs-nonexisting"'
  );
});
