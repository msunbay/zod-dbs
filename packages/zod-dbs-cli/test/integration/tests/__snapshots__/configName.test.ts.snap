// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI accepts custom config file name 1`] = `
"
zod-dbs CLI

- provider: test-provider (Test Provider)
- outputDir: ./test/integration/output/customConfigFile
- include: users,posts
- exclude: ^temp_
- zodVersion: 4
- stringifyDates: true
- defaultEmptyArray: true
- caseTransform: false
- configName: test

"
`;
